sdk/Lusid.Sdk.Tests/tutorials/Instruments/DemoEquityOption.cs

```diff
@@ -0,0 +1,181 @@
+        public void DemoEquityOptionCashFlows()
+        {
+            var scope = "DemoEquityOptionCashFlows";
+
+            // upsert option
+            var option = InstrumentExamples.CreateExampleEquityOption();
+            string uniqueId = "id-equityOption-1";
+            var response = _testDataUtilities.UpsertOtcToLusid(option, "some-name-for-this-equityOption", uniqueId);
+            var luid = response.Values.First().Value.LusidInstrumentId;
+
+            // for equity option cashflows, we need the following market data to determine intrinsic value
+            _testDataUtilities.CreateAndUpsertSimpleQuote(scope, "ACME", QuoteSeriesId.InstrumentIdTypeEnum.RIC, 110m, "USD", _demoEffectiveAt);
+
+            // create a new portfolio and add the option to it via a transaction
+            var portfolioCode = _testDataUtilities.CreateTransactionPortfolio(scope);
+            var reqs = new List<TransactionRequest> {_testDataUtilities.BuildTransactionRequest(luid, 100m, 5m, "USD", option.StartDate, "Buy")};
+            _transactionPortfoliosApi.UpsertTransactions(scope, portfolioCode, reqs);
+            var holdings = _transactionPortfoliosApi.GetHoldings(scope, portfolioCode);
+            Assert.That(holdings.Values.Count == 2); // one holding for the option, and an opposite holding of cash
+
+            // create a recipe to tell lusid where to find the requisite market data
+            // we require a model to estimate/determine future cashflows (for physically settled options, we currently assume exercise in all models)
+            // we choose ConstantTimeValueOfMoney since it has the fewest dependencies
+            var recipeCode = "EquityOption_CashFlowsRecipe";
+            var pricingOptions = new PricingOptions(new ModelSelection(ModelSelection.LibraryEnum.Lusid, ModelSelection.ModelEnum.ConstantTimeValueOfMoney));
+            var recipe = new ConfigurationRecipe
+            (
+                scope: scope,
+                code: recipeCode,
+                market: new MarketContext(
+                    new List<MarketDataKeyRule> { },
+                    options: new MarketOptions(defaultSupplier: "Lusid", defaultScope: scope, defaultInstrumentCodeType: "RIC")
+                ),
+                pricing: new PricingContext(options: pricingOptions)
+            );
+            _testDataUtilities.UpsertRecipe(recipe);
+
+            var cashflows = _transactionPortfoliosApi.GetPortfolioCashFlows(scope, portfolioCode, effectiveAt: _demoEffectiveAt,
+                windowStart: option.StartDate.AddDays(-3), windowEnd: option.OptionMaturityDate.AddDays(3),
+                recipeIdScope: scope, recipeIdCode: recipeCode).Values;
+            Assert.That(cashflows.Count, Is.EqualTo(1));
+            var cashflow = cashflows[0];
+            Assert.That(cashflow.Amount, Is.Negative);
+            Console.WriteLine($"Computed cash flow of {cashflow.Amount} {cashflow.Currency} at time {cashflow.PaymentDate}");
```

Comment:
Debating if one should then upsert the (transaction) cashflow back into lusid.

Reply:
Not sure, why would we want to do that?

Reply:
It's kinda the point of this GetUpsertableCashflow endpoint.
It's optional in some sense as I done that in another test.

Oh wait, nvm, you're calling GetPortfolioCashFlow not GetUpsertableCashflow. (I read TransactionPortfolioApi and "inferred")